### ISSUE TITLE
Issue in ErrorMaintainer.php

### ISSUE THREAD

**Issue by drupol** [friends-of-phpspec/phpspec-code-coverage#18]
Hi,

I'm having this issue when building [my project](https://github.com/drupol/psrcas) using `composer` with `--prefer-lowest` flag.

The complete log:
```
$ composer install
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 88 installs, 0 updates, 0 removals
  - Installing ocramius/package-versions (1.4.2): Loading from cache
  - Installing symfony/polyfill-ctype (v1.13.1): Loading from cache
  - Installing symfony/yaml (v5.0.2): Loading from cache
  - Installing symfony/process (v5.0.2): Loading from cache
  - Installing symfony/options-resolver (v5.0.2): Loading from cache
  - Installing symfony/finder (v5.0.2): Loading from cache
  - Installing symfony/filesystem (v5.0.2): Loading from cache
  - Installing psr/event-dispatcher (1.0.0): Loading from cache
  - Installing symfony/event-dispatcher-contracts (v2.0.1): Loading from cache
  - Installing symfony/event-dispatcher (v5.0.2): Loading from cache
  - Installing psr/container (1.0.0): Loading from cache
  - Installing symfony/service-contracts (v2.0.1): Loading from cache
  - Installing symfony/dependency-injection (v5.0.2): Loading from cache
  - Installing symfony/polyfill-php73 (v1.13.1): Loading from cache
  - Installing symfony/polyfill-mbstring (v1.13.1): Loading from cache
  - Installing symfony/console (v5.0.2): Loading from cache
  - Installing psr/log (1.1.2): Loading from cache
  - Installing symfony/config (v5.0.2): Loading from cache
  - Installing seld/jsonlint (1.7.2): Loading from cache
  - Installing monolog/monolog (1.25.3): Loading from cache
  - Installing gitonomy/gitlib (v1.2.0): Loading from cache
  - Installing doctrine/collections (1.6.4): Loading from cache
  - Installing phpro/grumphp (v0.17.1): Loading from cache
  - Installing justinrainbow/json-schema (5.2.9): Loading from cache
  - Installing nikic/php-parser (v4.3.0): Loading from cache
  - Installing phpstan/phpstan (0.12.3): Loading from cache
  - Installing phpstan/extension-installer (1.0.3): Loading from cache
  - Installing localheinz/diff (1.0.1): Loading from cache
  - Installing ergebnis/json-printer (3.0.2): Loading from cache
  - Installing ergebnis/json-normalizer (0.10.1): Loading from cache
  - Installing ergebnis/composer-json-normalizer (2.0.1): Loading from cache
  - Installing localheinz/composer-normalize (2.0.0): Loading from cache
  - Installing psr/http-message (1.0.1): Loading from cache
  - Installing psr/http-client (1.0.0): Loading from cache
  - Installing psr/cache (1.0.1): Loading from cache
  - Installing composer/xdebug-handler (1.4.0): Loading from cache
  - Installing composer/semver (1.5.0): Loading from cache
  - Installing wearejust/grumphp-extra-tasks (2.2.1): Loading from cache
  - Installing squizlabs/php_codesniffer (3.5.3): Loading from cache
  - Installing phpstan/phpstan-deprecation-rules (0.12.0): Loading from cache
  - Installing jakub-onderka/php-parallel-lint (v1.0.0): Loading from cache
  - Installing pimple/pimple (v3.2.3): Loading from cache
  - Installing twig/twig (v3.0.0): Loading from cache
  - Installing friendsoftwig/twigcs (v3.2.2): Loading from cache
  - Installing symfony/stopwatch (v5.0.2): Loading from cache
  - Installing symfony/polyfill-php72 (v1.13.1): Loading from cache
  - Installing paragonie/random_compat (v9.99.99): Loading from cache
  - Installing symfony/polyfill-php70 (v1.13.1): Loading from cache
  - Installing php-cs-fixer/diff (v1.3.0): Loading from cache
  - Installing doctrine/lexer (1.2.0): Loading from cache
  - Installing doctrine/annotations (v1.8.0): Loading from cache
  - Installing friendsofphp/php-cs-fixer (v2.16.1): Loading from cache
  - Installing drupol/phpcsfixer-configs-php (1.0.19): Loading from cache
  - Installing drupol/php-conventions (1.6.10): Loading from cache
  - Installing psr/http-factory (1.0.1): Loading from cache
  - Installing php-http/message-factory (v1.0.2): Loading from cache
  - Installing nyholm/psr7 (1.2.1): Loading from cache
  - Installing nyholm/psr7-server (0.4.1): Loading from cache
  - Installing phpstan/phpstan-strict-rules (0.12.0): Loading from cache
  - Installing theseer/tokenizer (1.1.3): Loading from cache
  - Installing sebastian/version (2.0.1): Loading from cache
  - Installing sebastian/environment (4.2.3): Loading from cache
  - Installing sebastian/code-unit-reverse-lookup (1.0.1): Loading from cache
  - Installing phpunit/php-text-template (1.2.1): Loading from cache
  - Installing phpunit/php-token-stream (3.1.1): Loading from cache
  - Installing phpunit/php-file-iterator (2.0.2): Loading from cache
  - Installing phpunit/php-code-coverage (7.0.10): Loading from cache
  - Installing doctrine/instantiator (1.3.0): Loading from cache
  - Installing sebastian/recursion-context (3.0.0): Loading from cache
  - Installing sebastian/exporter (3.1.2): Loading from cache
  - Installing phpspec/php-diff (v1.1.0): Loading from cache
  - Installing sebastian/diff (3.0.2): Loading from cache
  - Installing sebastian/comparator (3.0.2): Loading from cache
  - Installing webmozart/assert (1.6.0): Loading from cache
  - Installing phpdocumentor/reflection-common (2.0.0): Loading from cache
  - Installing phpdocumentor/type-resolver (1.0.1): Loading from cache
  - Installing phpdocumentor/reflection-docblock (4.3.3): Loading from cache
  - Installing phpspec/prophecy (1.10.0): Loading from cache
  - Installing phpspec/phpspec (6.1.1): Loading from cache
  - Installing friends-of-phpspec/phpspec-code-coverage (v4.3.2): Loading from cache
  - Installing webmozart/path-util (2.3.0): Loading from cache
  - Installing thecodingmachine/safe (v0.1.16): Loading from cache
  - Installing infection/infection (0.15.0): Loading from cache
  - Installing symfony/var-exporter (v5.0.2): Loading from cache
  - Installing symfony/cache-contracts (v2.0.1): Loading from cache
  - Installing symfony/cache (v4.4.2): Loading from cache
  - Installing symfony/http-client-contracts (v2.0.1): Loading from cache
  - Installing symfony/http-client (v4.4.2): Loading from cache
Writing lock file
Generating autoload files
ocramius/package-versions:  Generating version class...
ocramius/package-versions: ...done generating version class
Watch out! GrumPHP is sniffing your commits!
phpstan/extension-installer: Extensions installed
> phpstan/phpstan-deprecation-rules: installed
> phpstan/phpstan-strict-rules: installed
```

Now `composer update --prefer-lowest`

```
$ composer update --prefer-lowest                                                                                                     Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 7 installs, 60 updates, 10 removals
  - Removing webmozart/path-util (2.3.0)
  - Removing theseer/tokenizer (1.1.3)
  - Removing thecodingmachine/safe (v0.1.16)
  - Removing symfony/event-dispatcher-contracts (v2.0.1)
  - Removing psr/event-dispatcher (1.0.0)
  - Removing phpdocumentor/type-resolver (1.0.1)
  - Removing phpdocumentor/reflection-common (2.0.0)
  - Removing ergebnis/json-printer (3.0.2)
  - Removing ergebnis/json-normalizer (0.10.1)
  - Removing ergebnis/composer-json-normalizer (2.0.1)
  - Downgrading ocramius/package-versions (1.4.2 => 1.2.0): Loading from cache
  - Downgrading symfony/yaml (v5.0.2 => v3.4.0): Loading from cache
  - Downgrading symfony/finder (v5.0.2 => v3.4.0): Loading from cache
  - Downgrading symfony/process (v5.0.2 => v3.4.0): Loading from cache
  - Downgrading symfony/event-dispatcher (v5.0.2 => v3.4.0): Loading from cache
  - Downgrading symfony/polyfill-mbstring (v1.13.1 => v1.0.0): Loading from cache
  - Downgrading psr/log (1.1.2 => 1.1.0): Loading from cache
  - Installing symfony/debug (v2.8.0): Loading from cache
  - Downgrading symfony/console (v5.0.2 => v3.4.17): Loading from cache
  - Downgrading symfony/filesystem (v5.0.2 => v3.4.0): Loading from cache
  - Downgrading justinrainbow/json-schema (5.2.9 => 5.2.0): Loading from cache
  - Downgrading symfony/options-resolver (v5.0.2 => v3.4.0): Loading from cache
  - Installing localheinz/json-printer (2.0.1): Loading from cache
  - Installing localheinz/json-normalizer (0.9.0): Loading from cache
  - Downgrading localheinz/diff (1.0.1 => 1.0.0): Loading from cache
  - Installing localheinz/composer-json-normalizer (1.0.2): Loading from cache
  - Downgrading localheinz/composer-normalize (2.0.0 => 1.3.1): Loading from cache
  - Downgrading symfony/dependency-injection (v5.0.2 => v3.4.0): Loading from cache
  - Downgrading symfony/config (v5.0.2 => v3.4.0): Loading from cache
  - Downgrading seld/jsonlint (1.7.2 => 1.1.0): Loading from cache
  - Downgrading monolog/monolog (1.25.3 => 1.16.0): Loading from cache
  - Downgrading gitonomy/gitlib (v1.2.0 => v1.0.3): Loading from cache
  - Downgrading doctrine/collections (1.6.4 => v1.2): Loading from cache
  - Downgrading phpro/grumphp (v0.17.1 => v0.16.2): Loading from cache
  - Downgrading phpstan/extension-installer (1.0.3 => 1.0): Loading from cache
  - Downgrading sebastian/version (2.0.1 => 2.0.0): Loading from cache
  - Downgrading sebastian/code-unit-reverse-lookup (1.0.1 => 1.0.0): Loading from cache
  - Downgrading phpunit/php-text-template (1.2.1 => 1.2.0): Loading from cache
  - Downgrading phpunit/php-file-iterator (2.0.2 => 1.3.2): Loading from cache
  - Downgrading sebastian/environment (4.2.3 => 2.0.0): Loading from cache
  - Downgrading phpunit/php-token-stream (3.1.1 => 1.4.2): Loading from cache
  - Downgrading phpunit/php-code-coverage (7.0.10 => 5.0.0): Loading from cache
  - Downgrading doctrine/instantiator (1.3.0 => 1.0.5): Loading from cache
  - Downgrading symfony/polyfill-ctype (v1.13.1 => v1.8.0): Loading from cache
  - Downgrading symfony/service-contracts (v2.0.1 => v1.1.1): Loading from cache
  - Downgrading symfony/polyfill-php73 (v1.13.1 => v1.11.0): Loading from cache
  - Downgrading sebastian/recursion-context (3.0.0 => 1.0.0): Loading from cache
  - Downgrading sebastian/exporter (3.1.2 => 1.2.0): Loading from cache
  - Downgrading phpspec/php-diff (v1.1.0 => v1.0.1): Loading from cache
  - Downgrading sebastian/diff (3.0.2 => 1.4.0): Loading from cache
  - Downgrading sebastian/comparator (3.0.2 => 1.2.3): Loading from cache
  - Downgrading webmozart/assert (1.6.0 => 1.3.0): Loading from cache
  - Installing dflydev/markdown (v1.0.0): Loading from cache
  - Downgrading phpdocumentor/reflection-docblock (4.3.3 => 2.0.0): Loading from cache
  - Downgrading phpspec/phpspec (6.1.1 => 5.1.2): Loading from cache
  - Downgrading friends-of-phpspec/phpspec-code-coverage (v4.3.2 => v4.3.0): Loading from cache
  - Downgrading pimple/pimple (v3.2.3 => v3.2.0): Loading from cache
  - Installing padraic/humbug_get_contents (1.0.4): Loading from cache
  - Installing padraic/phar-updater (1.0.4): Loading from cache
  - Downgrading composer/xdebug-handler (1.4.0 => 1.3.0): Loading from cache
  - Downgrading infection/infection (0.15.0 => 0.12.2): Loading from cache
  - Downgrading symfony/var-exporter (v5.0.2 => v4.2.0): Loading from cache
  - Downgrading symfony/cache-contracts (v2.0.1 => v1.1.1): Loading from cache
  - Downgrading symfony/cache (v4.4.2 => v4.3.0): Loading from cache
  - Downgrading symfony/http-client-contracts (v2.0.1 => v1.1.8): Loading from cache
  - Downgrading symfony/http-client (v4.4.2 => v4.4.0): Loading from cache
  - Downgrading symfony/stopwatch (v5.0.2 => v3.0.0): Loading from cache
  - Downgrading symfony/polyfill-php72 (v1.13.1 => v1.4.0): Loading from cache
  - Downgrading paragonie/random_compat (v9.99.99 => v1.0.0): Loading from cache
  - Downgrading symfony/polyfill-php70 (v1.13.1 => v1.0.0): Loading from cache
  - Downgrading doctrine/lexer (1.2.0 => v1.0): Loading from cache
  - Downgrading doctrine/annotations (v1.8.0 => v1.2.0): Loading from cache
  - Downgrading composer/semver (1.5.0 => 1.4.0): Loading from cache
  - Downgrading friendsofphp/php-cs-fixer (v2.16.1 => v2.16.0): Loading from cache
  - Downgrading drupol/phpcsfixer-configs-php (1.0.19 => 1.0.17): Loading from cache
  - Downgrading twig/twig (v3.0.0 => v1.41.0): Loading from cache
  - Downgrading php-http/message-factory (v1.0.2 => v1.0.0): Loading from cache
Writing lock file
Generating autoload files
phpstan/extension-installer: Extensions installed
> phpstan/phpstan-deprecation-rules: installed
> phpstan/phpstan-strict-rules: installed
Watch out! GrumPHP is sniffing your commits!
ocramius/package-versions:  Generating version class...
ocramius/package-versions: ...done generating version class
```

Running `phpspec` now.

```
$ ./vendor/bin/phpspec run --stop-on-failure -vvv                                                                                     14.7s  Sat 21 Dec 2019 06:22:39 PM CET

      drupol\psrcas\Cas

  51  ✔ can authenticate (932ms)
  94  ✔ can authenticate a request (2263ms)
 202  ✔ can authenticate a request in proxy mode (1385ms)
 301  ✔ can be constructed without base url (481ms)
 335  ✔ can check if the logger works during a failed authentication of service ticket (1314ms)
 382  ✔ can check if the logger works during a failed proxy validate request (719ms)
 413  ✔ can check if the logger works during a failed service validate request (722ms)
 444  ✔ can check if the logger works during a successful authentication of service ticket (740ms)
 475  ✔ can check if the request needs authentication (476ms)
 494  ✔ can detect when gateway and renew are set together (567ms)
 518  ✔ can detect wrong url (504ms)
 555  ✔ can do a request to validate a ticket (861ms)
 606  ✔ can handle proxy callback request (1068ms)
 698  ✔ can login (831ms)
 779  ✔ can logout (813ms)
 854  ✔ can parse a bad proxy request response (544ms)
 879  ✔ can parse a good proxy request response (539ms)
 904  ✔ can parse json in a response (578ms)
 941  ✔ can renew login (540ms)
 969  ✔ can request a proxy ticket (627ms)
1004  ✔ can validate a bad proxy ticket (657ms)
1056  ✔ can validate a bad service validate request (584ms)
1087  ✔ can validate a good proxy ticket (1534ms)
1192  ✔ can validate a good service validate request (1117ms)
1273  ✔ can validate a service ticket (915ms)
1308  ✔ can validate any type of ticket (623ms)
1372  ✔ is initializable (458ms)

      drupol\psrcas\Configuration\Properties

  13  ✔ can be used as an array (92ms)
  30  ✔ can modify the configuration (87ms)
  54  ✔ is initializable (85ms)

      drupol\psrcas\Handler\ProxyCallback

  20  ✔ can catch issue with the cache (817ms)
  54  ✔ can test if the cache is working (227ms)
  64  ✔ can test the logger when missing pgtId (154ms)
  80  ✔ can test the logger when missing pgtIou (157ms)
  96  ✔ can test the logger when no parameter is in the url (158ms)
 112  ✔ can test the logger when parameters are in the url (362ms)
 144  ✔ is initializable (131ms)

      drupol\psrcas\Introspection\Introspector

  18  ✔ can detect a proxy response (91ms)
  41  ✔ can detect a service validate response (117ms)
  79  ✔ can detect a service validate response with proxy (92ms)
 103  ✔ can detect a wrong response (108ms)
 132  ✔ can detect an authentication failure response (106ms)
 152  ✔ can detect an unknown type of response (79ms)
 175  ✔ can detect an unsupported parse format (76ms)
 192  ✔ is initializable (76ms)

      drupol\psrcas\Introspection\Proxy

  14  ✔ can detect a proxy response (121ms)

      drupol\psrcas\Introspection\ServiceValidate

  14  ✔ can detect a proxy service validate response (147ms)
  71  ✔ can detect a service validate response (141ms)

      drupol\psrcas\Redirect\Login

  20  ✔ can deal with renew and gateway parameters (160ms)
  55  ✔ can deal with renew parameter (138ms)
  82  ✔ can get a response (129ms)
  98  ✔ is initializable (102ms)

      drupol\psrcas\Redirect\Logout

  18  ✔ can get a response (142ms)
  34  ✔ is initializable (127ms)

      drupol\psrcas\Service\Proxy

  19  ✔ can detect a wrong proxy response (192ms)
  40  ✔ can detect when no credentials (123ms)
  49  ✔ is initializable (124ms)

      drupol\psrcas\Service\ProxyValidate

  24  ✔ can detect when no credentials (134ms)
  33  ✔ can get credentials with pgtUrl (295ms)
  80  ✔ can get credentials without pgtUrl (222ms)
 103  ✔ is initializable (124ms)

      drupol\psrcas\Utils\SimpleXml

  13  ✔ can convert xml from a string into an array (88ms)
  30  ✔ can convert xml into an array (89ms)
  56  ✔ is initializable (81ms)

      drupol\psrcas\Utils\Uri

  12  ✔ can check if some query parameters exists (155ms)
  34  ✔ can create a new uri with a new param (207ms)
  65  ✔ can create a new uri with new parameters (229ms)
  96  ✔ can get a single param from uri (121ms)
 115  ✔ can get query parameters (79ms)
 125  ✔ can remove parameters (184ms)
 148  ✔ can set multiple params at the same time (88ms)
 160  ✔ is initializable (77ms)

      tests\drupol\psrcas\Cas

  17  ✔ can test the proxy mode with pgtUrl (122ms)
  38  ✔ can test the proxy mode without pgtUrl (110ms)

      tests\drupol\psrcas\Service\ProxyValidate

  27  ✔ can check the visibility of some methods (400ms)
 117  ✔ can detect when no credentials (124ms)
 126  ✔ can log debugging information when trying to get unexisting pgtIou (159ms)
 156  ✔ can parse a response (154ms)
 176  ✔ is initializable (114ms)


14 specs
79 examples (79 passed)
33765ms

Generating code coverage report in html format ...

In ErrorMaintainer.php line 121:

  [PhpSpec\Exception\Example\ErrorException]
  warning: count(): Parameter must be an array or an object that implements Countable in /home/pol/dev/git/psrcas/vendor/phpunit/php-code-coverage/src/Report/Html/Renderer/File.php line 30
  7

Exception trace:
 PhpSpec\Runner\Maintainer\ErrorMaintainer->errorHandler() at n/a:n/a
 count() at /home/pol/dev/git/psrcas/vendor/phpunit/php-code-coverage/src/Report/Html/Renderer/File.php:307
 SebastianBergmann\CodeCoverage\Report\Html\File->renderSource() at /home/pol/dev/git/psrcas/vendor/phpunit/php-code-coverage/src/Report/Html/Renderer/File.php:61
 SebastianBergmann\CodeCoverage\Report\Html\File->render() at /home/pol/dev/git/psrcas/vendor/phpunit/php-code-coverage/src/Report/Html/Facade.php:117
 SebastianBergmann\CodeCoverage\Report\Html\Facade->process() at /home/pol/dev/git/psrcas/vendor/friends-of-phpspec/phpspec-code-coverage/src/Listener/CodeCoverageListener.php:104
 LeanPHP\PhpSpec\CodeCoverage\Listener\CodeCoverageListener->afterSuite() at /home/pol/dev/git/psrcas/vendor/symfony/event-dispatcher/EventDispatcher.php:212
 Symfony\Component\EventDispatcher\EventDispatcher->doDispatch() at /home/pol/dev/git/psrcas/vendor/symfony/event-dispatcher/EventDispatcher.php:44
 Symfony\Component\EventDispatcher\EventDispatcher->dispatch() at /home/pol/dev/git/psrcas/vendor/phpspec/phpspec/src/PhpSpec/Runner/SuiteRunner.php:66
 PhpSpec\Runner\SuiteRunner->run() at /home/pol/dev/git/psrcas/vendor/phpspec/phpspec/src/PhpSpec/Console/Command/RunCommand.php:178
 PhpSpec\Console\Command\RunCommand->execute() at /home/pol/dev/git/psrcas/vendor/symfony/console/Command/Command.php:255
 Symfony\Component\Console\Command\Command->run() at /home/pol/dev/git/psrcas/vendor/symfony/console/Application.php:964
 Symfony\Component\Console\Application->doRunCommand() at /home/pol/dev/git/psrcas/vendor/symfony/console/Application.php:248
 Symfony\Component\Console\Application->doRun() at /home/pol/dev/git/psrcas/vendor/phpspec/phpspec/src/PhpSpec/Console/Application.php:89
 PhpSpec\Console\Application->doRun() at /home/pol/dev/git/psrcas/vendor/symfony/console/Application.php:148
 Symfony\Component\Console\Application->run() at /home/pol/dev/git/psrcas/vendor/phpspec/phpspec/bin/phpspec:25
 {closure}() at /home/pol/dev/git/psrcas/vendor/phpspec/phpspec/bin/phpspec:27

run [-f|--format FORMAT] [--stop-on-failure] [--no-code-generation] [--no-rerun] [--fake] [-b|--bootstrap BOOTSTRAP] [--no-coverage] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-c|--config CONFIG] [--] <command> [<spec>]
``` 

Do you have any idea where it could come from ?

**Comment by jaylinski** [comment]
It seems the error you are getting was reported in issue https://github.com/sebastianbergmann/php-code-coverage/issues/551.

TL;DR: PHPUnit 5.x with PHP >= 7.2 will throw an error.

**Comment by drupol** [comment]
@jaylinski What would be the elegant way of fixing this? Tweaking composer.json ?

**Comment by jaylinski** [comment]
@drupol I'm afraid there is no elegant way to fix this. At least none I can think of.

**Comment by jaylinski** [comment]
Is this still an issue or can we close it?

**Comment by jaylinski** [comment]
@drupol Is this still an issue or can we close it?

**Comment by drupol** [comment]
No it's fine to close now. Thank.

**Comment by jaylinski** [comment]
@drupol You have to close yourself, I don't have the permissions. :sweat_smile:
